src/librustdoc/test.rs

```diff
@@ -273,13 +273,28 @@ fn runtest(test: &str, cratename: &str, cfgs: Vec<String>, libs: SearchPaths,
                     } else if count == 0 && compile_fail == true {
                         panic!("test compiled while it wasn't supposed to")
                     }
+                    if count > 0 && error_codes.len() > 0 {
+                        let out = String::from_utf8(data.lock().unwrap().to_vec()).unwrap();
```

Comment:
I think this may be the wrong approach for testing that error codes were emitted, we emit structured errors in the compiler, right? Couldn't we capture that and test against the error codes there directly?